### ISSUE TITLE
Accessibility Issue with Dropdown component

### ISSUE THREAD

**Issue by TianyuHou** [microsoft/fluentui#14379]
PWD Impact:

Visually challenged users who rely on screen readers are not able to change the dropdown selection details in the form field, as users are not able to expand the dropdown list boxes by pressing enter, when navigated to it in NVDA browse(down arrow key) or form mode(f).

 

Test Environment:

OS : Win 10 2004 (OS Build 19041.329)

Browser:  New Edge 83.0.478.58

Tools used: NVDA 2020.1

 

Repro Steps:

Step 1: Enable screen reader NVDA and open the UIFabric official controls websites in New Edge browser.

Step 2: dropdown section screen will be displayed.

Step 3: Now navigate to the dropdown list in browse or form mode and press enter and verify whether the combo box is getting expanded.

 

Actual Result:

Users are not able to expand the dropdown list boxes by pressing enter, when navigated to it in NVDA browse(down arrow key) or form mode(f).
Observation: Only when navigated in tab key and pressing enter the list boxes are getting expanded.

 

Expected Result:

Users should be able to expand the dropdown list boxes by pressing enter, when navigated to it in NVDA browse(down arrow key) or form mode(f).

**Comment by khmakoto** [comment]
Hello @TianyuHou, thanks for filing this issue! I can't seem to repro the behavior you indicated as I'm always able to expand the `Dropdown` component when navigating via the down arrow key in NVDA browse mode. Are there any other things I should be aware of to repro this behavior?

**Comment by TianyuHou** [comment]
Could you please try tthis link:  https://developer.microsoft.com/en-us/fluentui#/controls/web/dropdown

Almost all of our teammates have tried this link, and only muti-selected dropdown list could be navigated by down arrow key in NVDA browse mode. and I'm thinking is that because of the muti-selected dropdown has 'button' as the role, while the single selected dropdown used 'listbox' as role

**Comment by khmakoto** [comment]
I think I wasn't able to reproduce the bug because I was tabbing initially and then moved with the down arrow key, but I'm able to reproduce it now. However, from what I can see the component is following the ARIA spec correctly.

I've tried with other screen readers and this behavior does not reproduce. I also ran the Accessibility Insights tool on our website using the link you provided and no issues related to this were flagged either. Though this tool doesn’t flag all ARIA issues, I couldn’t find any other discrepancies between the spec and this component’s base code. As a result this isn’t actionable on our end and looks to be a NVDA specific bug - I suggest you file an issue on their repo: https://github.com/nvaccess/nvda.

Let us know if there is anything else we can do to help!

**Comment by TianyuHou** [comment]
Thanks for your reply, actually this bug was raised from Microsoft internal accessibility team to us. And we resolved it by replacing all the Dropdown components with Combobox which works greatly well with NVDA screen reader, so I'm just wondering what is the real issue under the hood

**Comment by khmakoto** [comment]
Well, the `Dropdown` and `Combobox` components are different because they serve similar but different purposes. However, each one should be doing the correct things with respect to aria regarding their scenarios.
